API/Features/Image/ImageController.cs

```diff
@@ -1,63 +1,342 @@
+
+        if (images.Members is { Length: > 0 })
+        {
+            if (BulkPatchMayCauseReprocessing(images))
+            {
+                return HydraProblem(
+                    "Bulk patching operations may not contain origin or image policy information.", 
+                    null, 400, "Not Supported", null);
+            }
+            
+            // Is it OK if a patched image has a different space? 
+            // Yes, I think this is OK, it's a Move operation.
+                
+            if (images.Members.Any(image => image.ModelId == null))
+            {
+                // And this ModelId is NOT the customer/space/id construct that the DB has for a primary key.
+                // It used to be... but we're not going to do that. hydraImage.ToDlcsModel() silently corrects this form.
+                    
+                return HydraProblem(
+                    "All assets must have a ModelId", 
+                    null, 400, "Missing identifier", null);
+            }
+            foreach (var hydraImage in images.Members)
+            {
+                try
+                {
+                    // Here a Hydra object is being passed into the MediatR layer.
+                    // Should it get converted to a DLCS Model Asset first?
+                    var asset = hydraImage.ToDlcsModel(customerId, spaceId);
+                    var request = new PutOrPatchImage { Asset = asset, Method = "PATCH" };
+                    var result = await mediator.Send(request);
+                    if (result.Asset != null)
+                    {
+                        patchedAssets.Add(result.Asset);
+                    }
+                    else
+                    {
+                        logger.LogError($"We did not get an asset back for {asset.Id}");
+                    }
+                }
+                catch (APIException apiEx)
+                {
+                    return HydraProblem(
+                        apiEx.Message, 
+                        null, 500, apiEx.Label, null);
+                }
+                catch (Exception ex)
+                {
+                    return HydraProblem(
+                        ex.Message, 
+                        null, 500, "Could not patch images", null);
+                }
+            }
         }
+
+        var urlRoots = getUrlRoots();
+        var output = new HydraCollection<DLCS.HydraModel.Image>
+        {
+            WithContext = true,
+            Members = patchedAssets.Select(a => a.ToHydra(urlRoots)).ToArray(),
+            TotalItems = patchedAssets.Count,
+            Id = Request.GetDisplayUrl() + "?patch_" + Guid.NewGuid()
+        };
+        return Ok(output);
+    }
+
+    private bool BulkPatchMayCauseReprocessing(HydraCollection<DLCS.HydraModel.Image> images)
+    {
+        if (images.Members == null) return false;
+        
+        // this should check the same things as AssetPreparer::PrepareAssetForUpsert

```

Comment:
Could `PrepareAssetForUpsert` be used and the result checked to see if it requires reingest?

Reply:
It could but we'd have to have acquired the existing Asset (rather than Hydra version).
I was conscious that there's a little bit of duplication here but it's specific to the batch patch, a quick check on the hydra objects before the next layer is invoked.